app/browser/api/ledger.js

```diff
@@ -2718,22 +2718,43 @@ const onPublisherTimestamp = (state, oldTimestamp, newTimestamp) => {
 }
 
 const checkReferralActivity = (state) => {
+  console.log('checkReferralActivity')
   const downloadId = updateState.getUpdateProp(state, 'referralDownloadId')
 
   if (!downloadId) {
     updater.checkForUpdate(false, true)
-    return
+    return state
   }
 
-  module.exports.roundtrip({
-    server: referralServer,
-    method: 'PUT',
-    path: '/promo/activity',
-    payload: {
-      download_id: downloadId,
-      api_key: referralAPI
-    }
-  }, {}, activityRoundTrip)
+  const timestamp = updateState.getUpdateProp(state, 'referralAttemptTimestamp') || 0
+  const count = updateState.getUpdateProp(state, 'referralAttemptCount') || 0
+
+  console.log('counter', count)
+
+  if (count >= 5) {
+    console.log('I gave up :(')
+    updater.checkForUpdate(false, true)
+    return state
+  }
+
+  if (new Date().getTime() - timestamp >= ledgerUtil.milliseconds.minute * 2) {
```

Comment:
you might consider capturing `new Date().getTime()` with a variable and re-using below (when updating state). This way the timestamp is exactly the same in both cases (which should prevent failures when testing, if timestamps don't match exactly)

Reply:
true, done